examples/map/map_segment.py

```diff
@@ -0,0 +1,88 @@
+"""
+=======================================================
+Segmenting a Map based on transformation of coordinates
+=======================================================
+
+A demostration creating a segment of a particular map from
+transformed coordinates.
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+import sunpy.map
+from sunpy.coordinates import NorthOffsetFrame
+from sunpy.data.sample import AIA_171_IMAGE
+
+###############################################################################
+# We start with the sample data.
+smap = sunpy.map.Map(AIA_171_IMAGE)
+
+###############################################################################
+# A utility function gives us access to the helioprojective coordinate of each
+# pixel. From this we then transform the coordinates to the required frame.
+all_hpc = sunpy.map.all_coordinates_from_map(smap)
+all_hgs = all_hpc.transform_to("heliographic_stonyhurst")
+
+###############################################################################
+# Let's then segment the data based on coordinates and create a boolean mask
+# where True indicates the segmented mask.
+segment = np.logical_and(all_hgs.lon < 10 * u.deg, all_hgs.lon > 0 * u.deg)
```

Comment:
I think we should start off doing the segmentation using the numpy convention of `True` being invalid data, rather than applying `np.logical_not` later on. So I would reverse the logic here.
```suggestion
###############################################################################
# Let's then segment the data based on coordinates and create a boolean mask
# where True indicates invalid or deselected data.
# This is the convention used by the `numpy.ma` module.
segment = np.logical_and(all_hgs.lon > 10 * u.deg, all_hgs.lon < 0 * u.deg)
```

Reply:
This doesn't really perform the segmentation. It produces a segment which is `False` completely.

Reply:
Does it need to be logical_or when you reverse the signs? I don't have time to experiment right now, but we should really try and build the logic the right way around. I think that would make it easier to understand.

Reply:
Yes, `logical_or()`

Reply:
But based on `False` being the masked arrays accepted value, I'm not sure if we can skip the `np.logical_not`. 
`np.logical_or` doesn't provide the right plot either :sweat_smile:

Reply:
![image](https://user-images.githubusercontent.com/50923538/112321047-98897500-8cd5-11eb-9e32-a1cb250954f6.png)
This works but I'm not too sure why the rest of the image isn't being as `all_hpc` only contains coordinates till 90deg? I'm not too sure about this

Reply:
You also need to mask out where `all_hgs.lon` is NaN